### PR TITLE
#15 Fix TypeError (String can't be coerced into Integer) on collect_output

### LINKED ISSUE #15: TypeError (String can't be coerced into Integer) on collect_output
steps to reproduce

```ruby
include Glueby::Contract::TxBuilder
include Glueby::Contract::Timestamp::Util

results = list_unspent
fee = 10_000
sum, outputs = collect_outputs(results, fee)
# => TypeError (String can't be coerced into Integer)
```

## Expect
return `sum` and `outputs`

## Actual
TypeError (String can't be coerced into Integer)

--

I found a bug at `collect_outputs` method.
`results` is `Hash`. `results[0]["amount"]` is `String`. But sum that is amount calculation use String and Integer.

That cause TypeError.
`"50.0000" * 10_000` may be like `50.000050.0000...`.
`0 + "50.0000"` cause TypeError.

I will fix this issue.
I will create pull request asap.

### PR DESCRIPTION
**Pull request by JunOkumura**
cast amount from String to Integer.
The amount is like "50.000" that is String and TPC digit.

resolve #15

### CODE REVIEW

File: lib/glueby/contract/tx_builder.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ module TxBuilder
       # @raise [InsufficientFunds] if result of listunspent is not enough to pay the specified amount
       def collect_outputs(results, amount)
         results.inject([0, []]) do |sum, output|
-          new_sum = sum[0] + (output['amount'] * 100_000_000)
+          new_sum = sum[0] + (output['amount'].to_i * 100_000_000)
```
Comment (azuchi) [inline lib/glueby/contract/tx_builder.rb:11]: The values returned by listunspent include values after the decimal point, such as `"0.41547730"`. 
In that case, this code output `0` value.

And please implement unit test.

Comment (JunOkumura) [reply lib/glueby/contract/tx_builder.rb:11]: OK, Thank you review.
I will fix it, and run unit tests.

**Comment by JunOkumura** [comment]
I fixed it is reviewed code by you.

I modified code in `lib/glueby/internal/wallet/tapyrus_core_wallet_adapter.rb` for using `tpc_to_tapyrus`.
`tpc_to_tapyrus` move to module that is `Glueby::Internal::Wallet::TapyrusCoreWalletAdapter::Util`, and it change method from private to public.

I also fixed some test datas.
I returned the amount from rpc is String, but test data was Float.
I replaced them to String.

File: lib/glueby/contract/tx_builder.rb [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+
       # collect outputs in results so that sum of them exceeds up to the specified amount.
       # @param [Array] results of listunspent
       # @param [Numeric] amount
       # @return [(Numeric, Array)] sum value of outputs and outputs
       # @raise [InsufficientFunds] if result of listunspent is not enough to pay the specified amount
       def collect_outputs(results, amount)
         results.inject([0, []]) do |sum, output|
-          new_sum = sum[0] + (output['amount'] * 100_000_000)
+          new_sum = sum[0] + tpc_to_tapyrus(output['amount'])
```
Comment (azuchi) [inline lib/glueby/contract/tx_builder.rb:13]: The conversion process itself should be done in `Timestamp::Util#listunspent`.

Comment (JunOkumura) [reply lib/glueby/contract/tx_builder.rb:13]: ok

**Comment by JunOkumura** [comment]
`tpc` and `tapyrus` translation moved to `Timestamp#list_unspent`.
So I do not need to change some mock amounts, revert it. But the amounts had to be translated to `tapyrus`, I replaced from `tpc` to `tapyrus`.

File: spec/glueby/contract/timestamp_spec.rb [hunk 3]
```diff
@@ -13,11 +13,11 @@
       [{
         'txid' => '0555c5af698db73ed6378d2ed3c71e45fc6a1dbbcf931a248a7c9221f1d3220c',
         'vout' => 0,
-        'amount' => 1.00000000
+        'amount' => 100000000
```
Comment (azuchi) [inline spec/glueby/contract/timestamp_spec.rb:16]: This data should be a string as it comes from the Bitcoin Core RPC `listunspent`.

Comment (JunOkumura) [reply spec/glueby/contract/timestamp_spec.rb:16]: I got it.
I will fix soon.

**Comment by JunOkumura** [comment]
I changed the amount in `timestamp_spec.rb` to String that is return of `Tapyrus Core RPC`.

**Pull request merged**